src/udev/udev-builtin-net_id.c

```diff
@@ -821,6 +823,13 @@ static int names_devicetree(sd_device *dev, const char *prefix, EventMode mode)
         ofnode_path--;
         assert(path_is_absolute(ofnode_path));
 
+        if (naming_scheme_has(NAMING_DEVICETREE_ETHERNET_PORTS)) {
+                /* Check if there are 'ethernet-ports' child nodes of ethernet */
+                r = sd_device_new_child(&eth_ports_dev, ofnode_dev, "ethernet-ports");
```

Comment:
This should also match `ports`.

https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/Documentation/devicetree/bindings/net/ethernet-switch.yaml#n45

Reply:
Thanks for pointhig this, I missed this 'ports' node.

By looking at the kernel `ethernet-switch.yaml` and devicetrees, looks like we need to use 'ethernet-ports' or 'ports', which means both at the same time or more than one is not allowed. Therefore, I updated the code and it will look for 'ethernet-ports' first and, if it fails, it will check for 'ports'.